tests/shared/core/test_domain.py

```diff
@@ -1611,6 +1611,26 @@ def test_domain_invalid_yml_in_folder():
         Domain.from_directory("data/test_domains/test_domain_from_directory1/")
 
 
+def test_domain_with_duplicates():
+    """
+    Check if a domain with duplicated slots, responses and intents contains
+    a correct information in `duplicates` field.
+    """
+    domain = Domain.from_directory("data/test_domains/test_domain_with_duplicates/")
+    assert domain.duplicates["slots"] == ["mood"]
+    assert domain.duplicates["responses"] == ["utter_greet", "utter_did_that_help"]
+    assert domain.duplicates["intents"] == ["greet"]
+
+
+def test_domain_duplicates_when_one_domain_file():
```

Comment:
Perhaps it would be good to also add a test for a valid domain dir, not only domain file?